app/carpool/views.py

```diff
@@ -165,17 +165,18 @@ def start_geojson():
 @pool_bp.route('/carpools/mine', methods=['GET', 'POST'])
 @login_required
 def mine():
+    carpools = {'future': [], 'past': []}
     # Start with carpools you're driving in
-    carpools = current_user.get_driving_carpools().all()
+    for carpool in current_user.get_driving_carpools():
+        carpools['future' if carpool.future else 'past'].append(carpool)
 
     # Add in carpools you have ride requests for
-    carpools.extend([
-        req.carpool
-        for req in current_user.get_ride_requests_query()
-    ])
+    for req in current_user.get_ride_requests_query():
+        carpools['future' if req.carpool.future else 'past'].append(req.carpool)
 
     # Then sort by departure date
-    carpools.sort(key=lambda c: c.leave_time)
+    carpools['future'].sort(key=lambda c: c.leave_time)
+    carpools['past'].sort(key=lambda c: c.leave_time)
 
```

Comment:
I think we can move all of this logic into the model and it will clean this up a bit, especially w/ making testing easier. 

Something like:

```python
def mine():
    upcoming_carpools = current_user. upcoming_carpools()
    past_carpools = current_user.past_carpools()

    return render_template('carpools/mine.html', upcoming_carpools=upcoming_carpools, past_carpools=past_carpools)

```
and over in models

```python
class Person():
  ...
  ... 
  def upcoming_carpools(self):
     ...
 
  def past_carpools(self):
    ...
```

Reply:
I agree, @joearasin. Can you extract your comment into an issue so we don't lose track of it?